src/main/java/net/minecraftforge/items/IElytra.java

```diff
@@ -23,11 +23,19 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.ResourceLocation;
 
+/**
+ * Implementing this on an item will allow that item to act as a set of elytra.
+ * Any item implementing this must be able to be placed in the chest plate slot.
```

Comment:
Why does it need to be a chestplate that seems a bit restrictive could you change it so it works in any equipmentslot (for example elytra boots)

Reply:
As far as I know from looking at his patches, this PR is minimal and just replaces the vanilla itemstack and item check for something with extensible logic, instead of replacing the entire detection logic.

In my eyes this PR does exactly what it says on the tin.
And that is fine.
Another PR in the future can add the additional check that makes it possible from anywhere, let's start somewhere small and see how the community reacts.

Reply:
Because this essentially just extends the vanilla `item == Items.ELYTRA` check which only checks the chest plate slot. I could extend it to the other equipment slots but i was trying to minimize changes to vanilla code. And if i am extending it to the other equipment slots what about an item that lets you fly when you are holding it? Or some sort of charm that works as long as its in your inventory like an angel ring? What about things like baubles? Where do you draw the line? 

This is why i decided to keep it as simple as possible. This is where something like the elytra entity attribute that was attempted in #4958 would be useful as its is much more flexible. But its also a lot more complicated requiring much more invasive changes to vanilla code which is probably why it never got merged. 

People have been trying to get elytra support merged since 2018. I just want to create a simple pr that might actually get merged.